app/helpers/announcements_helper.rb

```diff
@@ -4,10 +4,11 @@ def announcement_hidden?(announcement)
   end
 
   def current_announcement
+    return @current_announcement if defined? @current_announcement
     @current_announcement = Announcement.current
```

Comment:
It is a memoization of the instance variable @current_annoucement, the reason `if defined?` is used is to get around `nil` which may happen if there isn't a **current* Announcement (one which has dates spanning `Date.today`